pkg/serverless/metrics/enhanced_metrics.go

```diff
@@ -301,6 +304,67 @@ func SendCPUEnhancedMetrics(userCPUOffsetMs, systemCPUOffsetMs float64, tags []s
 	})
 }
 
+func SendNetworkEnhancedMetrics(networkOffsetData *proc.NetworkData, tags []string, demux aggregator.Demultiplexer) {
+	networkData, err := proc.GetNetworkData(proc.ProcNetDevPath)
+	if err != nil {
+		log.Debug("Could not emit network enhanced metrics")
+		return
+	}
+
+	now := float64(time.Now().UnixNano()) / float64(time.Second)
+	GenerateNetworkEnhancedMetrics(GenerateNetworkEnhancedMetricArgs{
```

Comment:
I think you can combine these two methods into one (`SendNetworkEnhancedMetrics` and `GenerateNetworkEnhancedMetrics`) since the `GenerateNetworkEnhancedMetrics` method is only called from this one location. I'm guessing you did it this way in order to make testing easier? In which case, I'd recommend making the `GenerateNetworkEnhancedMetrics` not exported (ie `GenerateNetworkEnhancedMetrics`).

Even still, this is fine how it is, just a nit.

Reply:
Exactly, did it this way to make testing easier. Updated to unexport the method